### ISSUE TITLE
Need environment management system

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Sandia-OpenSHMEM/SOS#13]
```
Currently, the environment variables used in the code are checked (and worse, 
listed in sma_info) in an ad-hoc manner.  This is probably not the best path 
forward; we really need some type of management infrastructure to make sure 
that sma_info is always up to date.
```

Original issue reported on code.google.com by `bwbarrett@gmail.com` on 21 Nov 2013 at 1:44

**Comment by jdinan** [comment]
Partially addressed by the env var update in #118.

**Comment by jdinan** [comment]
We also need input validation for env vars.  E.g. checks that value provided is greater or equal to zero.

**Comment by jdinan** [comment]
Fixed by #412
